### PR TITLE
Added Rendering to Scene

### PR DESCRIPTION
**Pull request by The-Spaghetti-Is-Ready**
## What problem does this pull request address?

This solves the need for entrypoint to have a render loop and any other initializations in Entrypoint. Scene class now serves as its own independent 'Engine' class,  decoupling Entrypoint from engine code.

## How does this pull request solve the problem?

This problem is solved by moving all of the implementation that was originally in Entrypoint and moving it to Scene so that there is an interface for creating Scenes rather than have the engine depend on the entrypoint for drawing and/or defining and declaring instances.

The callback functions now utilize a user pointer to the Scene so the callbacks can access the data in the Scene interface wihout needing an input manager (this is a quick solution that is not complete, but given the time constraints it was necessary).

## What testing has been performed?

- [x] Builds successfully
- [x] Runs successfully
- [x] Entrypoint lookin gooooooooooooooood
- [x] Scene functions as before

### CODE REVIEW

**Comment by The-Spaghetti-Is-Ready** [comment]
haha big diff, sorry guys

**Review by Minywire — DISMISSED** [review]
nice and clean :))

**Review by Minywire — APPROVED** [review]
red lines goneski

File: Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp [hunk 1]
```diff
@@ -57,20 +178,95 @@ void Scene::init()
...
-    accumulatedFrameTime += deltaTime;
-    
-    bbSystems.setLight(*renderEngine, ShaderType::Default, viewMatrix);
-    Systems::drawTerrain(bbECS, ShaderType::Terrain, *renderEngine, resources.getSceneTerrain(), false, projectionMatrix, viewMatrix);
-    Systems::drawModels(bbECS, ShaderType::Default, *renderEngine, resources.getSceneModels(), projectionMatrix, viewMatrix);
+    while (!window.GetShouldClose()) {
+        auto current_frame = static_cast<float>(glfwGetTime());
+        float delta = current_frame - last_frame;
+        last_frame = current_frame;
+        accumulatedFrameTime += delta;
```
Comment (Idraliski) [inline Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp:187]: can we please change delta to be deltaTime, avoids merge conflicts (maybe) and adds more clarity to what "delta" is

Comment (The-Spaghetti-Is-Ready) [reply Bottlebrush_Engine/Bottlebrush_Engine/BBScene/Scene.cpp]: image

**Review by Idraliski — APPROVED** [review]

**Review by Minywire — APPROVED** [review]

**Pull request merged**